cabal-install/tests/UnitTests/Distribution/Solver/Modular/RetryLog.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE StandaloneDeriving #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module UnitTests.Distribution.Solver.Modular.RetryLog (
+  tests
+  ) where
+
+import Distribution.Solver.Modular.Message
+import Distribution.Solver.Modular.RetryLog
+import Distribution.Solver.Types.Progress
+
+import Test.Tasty (TestTree)
+import Test.Tasty.HUnit (testCase, (@?=))
+
+type Log a = Progress a String String
+
+tests :: [TestTree]
+tests = [
+    testCase "convert to and from RetryLog ending in failure" $
```

Comment:
I converted all of the tests to Quickcheck properties, except for the two that need to use `Message` as the step type.